modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,262 @@
+                    attribute_proto.graphs_size() > 0) {
+        CV_Error(Error::StsNotImplemented, "Unexpected attribute type");
+    }
+  }
+  return lp;
+}
+
+void ONNXImporter::populateNet(Net dstNet)
+{
+    CV_Assert(model_proto.has_graph());
+
+    onnx::GraphProto graph_proto = model_proto.graph();
+    std::map<std::string, Mat> weights = getWeights(graph_proto);
+    std::map<std::string, Mat>::iterator weight;
+
+    int layersSize = graph_proto.node_size();
+    LayerParams layerParams;
+    onnx::NodeProto node_proto;
+
+    for(int i = 0; i < layersSize; i++) {
+        node_proto = graph_proto.node(i);
+        layerParams = getLayerParams(node_proto);
+        if (node_proto.has_name() && !node_proto.name().empty()) {
+            layerParams.name = node_proto.name();
+        } else {
+            std::ostringstream s;
+            s << node_proto.op_type() << "_" << i;
+            layerParams.name = s.str();
+         }
+         std::string layer_type = node_proto.op_type();
+
+         if (layer_type == "MaxPool") {
+            layerParams.type = "Pooling";
+            layerParams.set("pool", "MAX");
+        } else if (layer_type == "LRN") {
+            layerParams.type = "LRN";
+            if (layerParams.has("size")) {
+                layerParams.set("local_size", layerParams.get<int>("size"));
+                layerParams.erase("size");
+            }
+        } else if (layer_type == "Gemm") {
+            layerParams.type = "InnerProduct";
+            for (int i = 1; i < node_proto.input_size(); i++) {
+                weight = weights.find(node_proto.input(i));
+                CV_Assert(weight != weights.end());
+                layerParams.blobs.push_back(weight->second);
+            }
+            layerParams.set("num_output", layerParams.blobs[0].size[0]);
+            int numLayerWeights = node_proto.input_size() - 1;
+            if (numLayerWeights == 2) {
+                layerParams.set("bias_term", true);
+            } else {
+                layerParams.set("bias_term", false);
+            }
+         } else if (layer_type == "Conv") {
+            layerParams.type = "Convolution";
+            for (int i = 1; i < node_proto.input_size(); i++) {
+                weight = weights.find(node_proto.input(i));
+                CV_Assert(weight != weights.end());
+                layerParams.blobs.push_back(weight->second);
+            }
+            layerParams.set("num_output", layerParams.blobs[0].size[0]);
+            int numLayerWeights = node_proto.input_size() - 1;
+            if (numLayerWeights == 2) {
+                layerParams.set("bias_term", true);
+            } else {
+                layerParams.set("bias_term", false);
+            }
+        } else if (layer_type == "Reshape") {
+            layerParams.type = "Reshape";
+            for (int i = 1; i < node_proto.input_size(); i++) {
+                weight = weights.find(node_proto.input(i));
```

Comment:
We need to name it somehow differently because it's not weights mat for this layer. And we don't need a loop here because Reshape takes only input blob and shape, doesn't it?

Reply:
done